host.py

```diff
@@ -292,20 +293,45 @@ def copy_to(self, src_file: str, dst_file: str) -> None:
     def need_sudo(self) -> None:
         self.sudo_needed = True
 
-    def run(self, cmd: str, log_level: int = logging.DEBUG, env: dict[str, str] = os.environ.copy()) -> Result:
+    def run(
+        self,
+        cmd: str,
+        log_level: int = logging.DEBUG,
+        env: Optional[dict[str, str]] = None,
+        *,
+        env_extra: Optional[Mapping[str, Optional[str]]] = None,
+    ) -> Result:
```

Comment:
This style is bad. All the args should be on one line. If the line is too long, the code needs to be refactored. It's a red flag.

Reply:
see https://github.com/bn222/cluster-deployment-automation/pull/133#discussion_r1565634324